### ISSUE TITLE
Documenting file formats supported by napari ecosystem

### ISSUE THREAD

**Issue by junxini** [napari/napari#4615]
## 🚀 Feature
 Prior to opening a file, I want to know which file formats are supported, so that I know whether the napari ecosystem can be used to view my image

## Current behavior (0.4.16)
There’s a lack of documentation that guides the user through the image loading step and informs them of formats that are supported by napari

## Pitch

- [ ] Clear instructions on Open File options (if there's more than 1)

- [ ] Clarity on how drag & drop files are handled

- [ ] A full list of file formats supported by the napari ecosystem broken down by:

  - formats supported by napari natively
  - formats supported by the plugins installed by the user
  - formats supported by plugins NOT installed by the user
  - Note: this list could be displayed in a new tab of `napari info` or under the `plugins` menu. Natively supported file formats should also be specified on the website

## Additional context

This feature request is under epic https://github.com/napari/napari/issues/4611, in collaboration with @DragaDoncila, @chili-chiu, and Lisa

**Comment by Czaki** [comment]
This requires dropping npe1 plugins.

**Comment by alisterburt** [comment]
Thanks for this @junxini - to clarify

>  A full list of file formats supported by the napari ecosystem broken down by:

What do you mean exactly by 'a full list', where do you want this to live? A full list of everything NOT installed by the user will likely become huge - I can imagine this living in the docs somewhere as a reference but can't see it living in napari itself in a particularly useful way

**Comment by DragaDoncila** [comment]
> This requires dropping npe1 plugins.

With the shim work we should be able to discern file formats supported by legacy plugins as well as new `npe2` ones, albeit maybe with a lesser degree of confidence.

> What do you mean exactly by 'a full list', where do you want this to live?

At a minimum this should be a list of file formats supported by `builtins`, and would live on the website. Currently this should be everything on [this list](https://imageio.readthedocs.io/en/stable/formats/index.html), and maybe it would be as easy as just linking out to this list. I do think this should be relatively front and centre on the website - it's strange that right now there's nowhere you can go to find out what we can read.

In terms of the full ecosystem including plugins, we will soon support filtering by file extensions on the hub, but we may also want to provide a dedicated search functionality for formats on the hub. This will also be supported by #4614

**Comment by chili-chiu** [comment]
> At a minimum this should be a list of file formats supported by `builtins`, and would live on the website. Currently this should be everything on [this list](https://imageio.readthedocs.io/en/stable/formats/index.html), and maybe it would be as easy as just linking out to this list. I do think this should be relatively front and centre on the website - it's strange that right now there's nowhere you can go to find out what we can read.
> 

@DragaDoncila is it accurate to say that napari supports file formats included in the imageio library plus Numpy-like arrays (e.g. Zarr)? Are there other reader libraries napari use? 

Also, is [the image layer page](https://napari.org/howtos/layers/image.html) the right place to include this information?

**Comment by DragaDoncila** [comment]
@chili-chiu yes that is correct, you would just add CSV, npy and zarr to that list. There are no other reader functions that we currently use, so this would be a comprehensive listing
